velox/exec/Task.cpp

```diff
@@ -564,8 +578,15 @@ void Task::start(
         "concurrentSplitGroups parameter must be greater then or equal to 1");
 
     uint32_t numPipelines;
+    std::shared_ptr<const core::PartitionedOutputNode> partitionedOutputNode{
+        nullptr};
+    int numOutputDrivers{0};
     {
       std::unique_lock<std::mutex> l(self->mutex_);
+      // NOTE: the driver might have been terminated by async external event
+      // even before it gets started.
+      CHECK_EARLY_TASK_ERROR();
```

Comment:
Not your code, but this method grew very large and hard to follow over time. It would be nice to refactor it first before introducing more logic.